worker/producer.go

```diff
@@ -15,8 +17,25 @@ type Producer struct {
 	quit          chan chan error
 }
 
-func NewProducer(queue *convoyRedis.RedisQueue) *Producer {
-	consumer := queue.Consumer()
+func NewProducer(cfg config.Configuration, eventQueue queue.Queuer) *Producer {
+	var consumer taskq.QueueConsumer
+	var queue queue.Queuer
+
+	if cfg.Queue.Type == config.RedisQueueProvider {
```

Comment:
@ogbanugot This should be wrapped into the `Queuer` interface.

Reply:
This is not very clear.